ci/travis/before_install-linux.sh

```diff
@@ -41,8 +41,16 @@ export PATH=$HOME/.local/bin:$PATH
 export PYTHONPATH=$HOME/.local/lib/python2.7/site-packages:$PYTHONPATH
 
 echo "Installing latest pip"
-pip install --ignore-installed --user setuptools
-pip install --ignore-installed --user pip
+pip install --upgrade --ignore-installed --user setuptools
+pip install --upgrade --ignore-installed --user pip
+
+# Print diagnostic info to help debug why we're not picking up the latest pip
+pip --version
+echo "WHICH PIP: $(which pip)"
+echo "WHEREIS PIP: $(whereis pip)"
+/usr/local/bin/pip --version
+/usr/local/bin/pip2.7 --version
+echo "PATH=$PATH"
```

Comment:
@breznak, It's still a problem that needs to be solved. pip complains that the current version 6.x and the latest is 8.x. The `--upgrade` downloads the 8.x wheel and "installs", then prints "successfully installed 6.x" (forgot the exact x). Any idea how to really fix this?